Day-15_Classes/classes.js

```diff
@@ -0,0 +1,154 @@
+// Exercises Level 1
+
+// Create an Animal class. The class will have name, age, color, legs properties and create different methods
+class Animal {
+  constructor (name, age, color, legs) {
+    this.name = name
+    this.age = age
+    this.color = color
+    this.legs = legs  
+  }
+
+  get getName(){
+    return this.name
+  }
+
+  get getAge(){
+    return this.age
+  }
+
+  get getColor(){
+    return this.color
+  }
+
+  get getLegs(){
+    return this.legs
+  }
+
+  eat() {
+    console.log('Nyam nyam')
+  }
+
+  sleep() {
+    console.log(' ... zzzZZZ')
+  }
+
+  getFullInfo() {
+    const fullInfo = `This anymal's name is ${this.getName}. It is ${this.getAge} years old and of ${this.getColor} color. It has ${this.getLegs} legs`
+    return fullInfo
+  }
+}
+
+
+// Create a Dog and Cat child class from the Animal Class.
+class Dog extends Animal {
+  saySomething() {
+    console.log('WOOF WOOF')
+  }
+}
+
+class Cat extends Animal {
+  saySomething() {
+    console.log('MEOW MEOW')
+  }
+}
+
+
+// Exercises Level 2
+
+// Override the method you create in Animal class
+class SmallDog extends Dog {
+  constructor(name, age, color, legs){
+    super(name, age, color, legs)
+  }
+
+  saySomething() {
+    console.log('Yip Yip')
+  }
+}
+
+
+// Exercises Level 3
+
+// Let's try to develop a program which calculate measure of central tendency of a sample(mean, median, mode) and measure of variability(range, variance, standard deviation). 
+// In addition to those measures find the min, max, count, percentile, and frequency distribution of the sample. 
+// You can create a class called Statistics and create all the functions which do statistical calculations as method for the Statistics class. Check the output below.
+const ages = [31, 26, 34, 37, 27, 26, 32, 32, 26, 27, 27, 24, 32, 33, 27, 25, 26, 38, 37, 31, 34, 24, 33, 29, 26]
+class Statistics {
+  constructor(data = []){
+  this.data = data
+  }
+
+  setData(newData) {
+    this.data = newData.sort((a,b) => a - b)
+  }
+
+  count(){
+    return this.data.length
+  }
+
+  sum() {
+    return this.data.reduce((acc, curr) => acc + curr,0)
+  }
+
+  min(){
+    return this.data[0]
```

Comment:
If you call this method before the `setData` method, it will return the first element in the `this.data` array or undefined if the array is empty. <br>
It's better, first, check if array is not empty. If it is, return undefined. If not, use `Max.min()` to find the minimum value in the array then return it.

Reply:
Updated with error handling for empty array and with Math.min() usage